tap_github/streams.py

```diff
@@ -158,6 +158,113 @@ def get_child_context(self, record: dict, context: Optional[dict]) -> dict:
     ).to_dict()
 
 
+class EventsStream(GitHubStream):
+    """
+    Defines 'Events' stream.
+    Issue events are fetched from the repository level (as opposed to per issue)
+    to optimize for API quota usage.
+    """
+
+    name = "events"
+    path = "/repos/{org}/{repo}/events"
+    primary_keys = ["id"]
+    replication_key = "created_at"
+    parent_stream_type = RepositoryStream
+    state_partitioning_keys = ["repo", "org"]
+    ignore_parent_replication_key = False
+
+    def get_records(self, context: Optional[dict] = None) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        if context and context.get("events", None) == 0:
+            self.logger.debug(f"No events detected. Skipping '{self.name}' sync.")
+            return []
+
+        return super().get_records(context)
```

Comment:
A little duplicative of the issueevents stream #2 . Maybe we could consider adding a utility py? Hopefully upstream would be ok with this.